docu/Concepts/TechnicalRequirements/UC_Introduction_of_Gradido-ID.md

```diff
@@ -0,0 +1,141 @@
+| id              | int    | the technical key of a contact entity                                                                                                                                  |
+| type            | int    | Defines the type of contact entry as enum: Email, Phone, etc                                                                                                           |
+| userID          | int    | Defines the foreign key to the `Users` table                                                                                                                         |
+| email           | String | defines the address of a contact entry of type Email                                                                                                                   |
+| phone           | String | defines the address of a contact entry of type Phone                                                                                                                   |
+| contactChannels | String | define the contact channel as comma separated list for which this entry is confirmed by the user e.g. main contact (default), infomail, contracting, advertisings, ... |
+
+### Database-Migration
+
+After the adaption of the database schema and to keep valid consistent data, there must be several steps of data migration to initialize the new and changed columns and tables.
+
+#### Initialize GradidoID
+
+In a one-time migration create for each entry of the `Users `tabel an unique UUID (version4).
+
+#### Primary Email Contact
+
+In a one-time migration read for each entry of the `Users `table the `Users.id` and `Users.email` and create for it a new entry in the `UsersContact `table, by initializing the contact-values with:
+
+* id = new technical key
+* type = Enum-Email
+* userID = `Users.id`
+* email = `Users.email`
+* phone = null
+* usedChannel = Enum-"main contact"
+
+and update the `Users `entry with `Users.emailId = UsersContact.Id` and `Users.passphraseEncryptionType = 1`
+
+After this one-time migration the column `Users.email` can be deleted.
+
+### Adaption of BusinessLogic
+
+The following logic or business processes has to be adapted for introducing the Gradido-ID
+
+#### Read-Write Access of Users-Table especially Email
+
+The ORM mapping has to be adapted to the changed and new database schema.
+
+#### Registration Process
+
+The logic of the registration process has to be adapted by
+
+* initializing the `Users.userID` with a unique UUID
+* creating a new `UsersContact `entry with the given email address and *maincontact* as `usedChannel `
+* set `emailID `in the `Users `table as foreign key to the new `UsersContact `entry
+* set `Users.passphraseEncrpytionType = 2` and encrypt the passphrase with the `Users.userID` instead of the `UsersContact.email`
```

Comment:
I think its worth to think about removing this salt altogether and replace it with a server-wide salt. This is a bit less security, but might be worthwhile to not be depending on a per-user-key for encryption. It might even be that our encryption of passwords have a salt already baked in by the sodium library and the whole salt thing is not a security feature at all. This has to be researched tho. I believe that if we change the password encryption algorithm, we should think it from scratch again and not just adjust some parameters. It might be that our current method is completely insecure - if we have a look, just dive a bit deeper and consider current web standards and best practises.

Reply:
i think there is a missunderstanding about the attribute passphraseEncryptionType.

This attribut will only distigue between the 
- encryption of the password is done by the email or 
- will be done by the Users.gradidoID, which will be the UUID.

In future there will be perhaps an other passphraseEncryptionType, which use the UUID but with a new encryption algorithm. 
Which input attributes and values for a password encryption are used can be detected by the value and the definition behind the value of a passphraseEncryptionType and is not an input value for encryption itself

Reply:
I fully understand the concept of `passphraseEncryptionType`. What I suggest is to really think about the new version of password encryption which you describe with `will be done by the Users.gradidoID, which will be the UUID.`. We should not jump to conclusions how to do this in the future. We should evaluate current web-standards and compare it to our current process and then make a decision how to change the password algo.